test/unit/definitions/divorce/AuthorisationCaseState.test.js

```diff
@@ -1,52 +1,34 @@
 const { uniqWith } = require('lodash');
-const fs = require('fs');
 
 const load = require;
-const authCaseState = Object.assign(require('definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState'), {});
+const authCaseStateCommon = Object.assign(require('definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState'), {});
 
 function isDuplicated(field1, field2) {
-  return field1.CaseTypeID === field2.CaseTypeID
-    && field1.CaseStateID === field2.CaseStateID
-    && field1.UserRole === field2.UserRole;
+    return field1.CaseTypeID === field2.CaseTypeID
+        && field1.CaseStateID === field2.CaseStateID
+        && field1.UserRole === field2.UserRole;
 }
 
-function mergeJsonFilesFor(whatFolder, nonProdStartsWith) {
-  const files = fs.readdirSync(whatFolder);
-  const authCaseStateNonProd = [];
+function mergeJsonFilesFor(whatEnvs) {
+    const authCaseForSpecificEnvs = Object
+        .assign(load(`definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-${whatEnvs}`), {});
 
-  files.forEach(filename => {
-    const filenameRegEx = new RegExp(`${nonProdStartsWith}.*.json`, 'i');
-    if (filename.match(filenameRegEx)) {
-       const nonProdFile = Object
-          .assign(load(`${whatFolder}/${filename}`), {});
-       authCaseStateNonProd.push(...nonProdFile);
-    }
-  });
-
-  return [...authCaseState, ...authCaseStateNonProd];
+    return [...authCaseStateCommon, ...authCaseForSpecificEnvs];
 }
 
 describe('AuthorisationCaseState', () => {
-  if (fs.existsSync('definitions/divorce/json/AuthorisationCaseState/')){
-     it('should contain a unique case state ID, case type ID and role (no duplicates) for non prod files', () => {
-        const jsonConfigFiles = mergeJsonFilesFor('definitions/divorce/json/AuthorisationCaseState/', 'AuthorisationCaseState-');
-        const uniqResult = uniqWith(jsonConfigFiles, isDuplicated);
-
-        expect(uniqResult).to.eql(jsonConfigFiles);
-     });
-  }
-
-  it('should contain a unique case state ID, case type ID and role (no duplicates) for prod file', () => {
-    const uniqResult = uniqWith(
-      authCaseState,
-      (field1, field2) => {
-        return field1.CaseTypeID === field2.CaseTypeID
-                   && field1.CaseStateID === field2.CaseStateID
-                   && field1.UserRole === field2.UserRole;
-      }
-    );
-
-    expect(uniqResult).to.eql(authCaseState);
-  });
+    it('should contain a unique case state, case type ID and role (no duplicates) for nonprod files', () => {
+        const nonProd = mergeJsonFilesFor('nonprod');
+        const uniqResult = uniqWith(nonProd, isDuplicated);
+
+        expect(uniqResult).to.eql(nonProd);
+    });
+
+    it('should contain a unique case state ID, case type ID and role (no duplicates) for prod file', () => {
+        const prodOnly = mergeJsonFilesFor('prod');
+        const uniqResult = uniqWith(prodOnly, isDuplicated);
+
+        expect(uniqResult).to.eql(prodOnly);
+    });
```

Comment:
Test suite updated as adding the -prod / -nonprod files in this folder was causing duplicate errors.